app/queries/servers/group.ts

```diff
@@ -36,6 +37,22 @@ export const queryGroupsByNameInChannel = (database: Database, name: string, cha
     );
 };
 
+export const queryGroupMembershipForUser = (database: Database, userId: string) => {
+    return database.collections.get<GroupMembershipModel>(GROUP_MEMBERSHIP).query(
+        Q.where('user_id', userId),
+    );
+};
+
 export const prepareGroups = (operator: ServerDataOperator, groups: Group[]) => {
     return operator.handleGroups({groups, prepareRecordsOnly: true});
 };
+
+export const prepareGroupMembershipsForUser = async (operator: ServerDataOperator, groups: Group[] | GroupModel[], userId: string) => {
+    const existingGroupMemberships = await queryGroupMembershipForUser(operator.database, userId).fetch();
+    existingGroupMemberships.forEach((gm) => gm.prepareDestroyPermanently());
+
+    const groupMembershipMap = groups.map((g) => ({id: `${userId}_${g.id}`, user_id: userId, group_id: g.id}));
+    const groupMemberships = await operator.handleGroupMemberships({groupMemberships: groupMembershipMap, prepareRecordsOnly: true});
```

Comment:
Just to make sure. Are we removing all groups and then adding them all? Shouldn't we just check the groups and update them, instead of adding / removing all of them every single time?

Reply:
I had a conversation with @shazm to have this done on the handler and to not delete them/insert them all